### PR TITLE
Allow parallel solve of models using ArcticDBConnector

### PR DESCRIPTION
**Pull request by dbrakenhoff**

### CODE REVIEW

File: pastastore/store.py [hunk 1]
```diff
@@ -1235,33 +1235,53 @@ def solve_models(
...
+            else self.conn,
             report=report,
             ignore_solve_errors=ignore_solve_errors,
             **kwargs,
         )
-        if self.conn.conn_type != "pas":
+        if self.conn.conn_type == "dict":
             parallel = False
             logger.error(
                 "Parallel solving only supported for PasConnector databases."
```
Comment (martinvonk) [inline pastastore/store.py:1258]: And ArcticDBConnector

**Comment by martinvonk** [comment]
Can't we pass initargs to the executor? To initialize the ArcticDB connector only `max_workers` times? Just like tqdm:
```python
with PoolExecutor(max_workers=max_workers, initializer=tqdm_class.set_lock,
 initargs=(lk,))
```

**Comment by martinvonk** [comment]
or do something like this:

```python
import shutil
with ThreadPoolExecutor(max_workers=4) as e:
    e.submit(shutil.copy, 'src1.txt', 'dest1.txt')
    e.submit(shutil.copy, 'src2.txt', 'dest2.txt')
    e.submit(shutil.copy, 'src3.txt', 'dest3.txt')
    e.submit(shutil.copy, 'src4.txt', 'dest4.txt')
```

```python
max_workers = 2
conns = [ArcticDBConnector() for conn in max_workers]
with ProcesPoolExecutor(max_workers=max_workers) as e:
    e.submit(solve_model, conn[0])
    e.submit(solve_model, conn[1])
```

**Comment by dbrakenhoff** [comment]
Good suggestion, I implemented parallel with initargs and an initializer for ArcticDBConnector.

This new approach adds a `_parallel()` method to the Connector objects which can be used to run other functions (staticmethods) in parallel as long as they store the result in the database (since nothing is returned). 

I had to move `ConnectorUtils` from `base.py` to `connectors.py` to ensure that the global connector per worker could be accessed by the `solve_model()` staticmethod. The _solve_model() staticmethod was moved to `ConnectorUtils` class. 

New results:
image

**Comment by martinvonk** [comment]
Wow pretty insane that you got this to work :) I'll review it tomorrow.

**Review by martinvonk — APPROVED** [review]
I didn't test this yet but looking through the code this seems like a good solution for arcticdb :)

**Pull request merged**